07-mongodb-mongoose/02-rest-api/controllers/products.js

```diff
@@ -0,0 +1,35 @@
+const Product = require('../models/Product');
+
+module.exports.productsBySubcategory = async function productsBySubcategory(ctx, next) {
+  try {
+    const {subcategory} = ctx.request.query;
+    const products = await Product.find(subcategory ? {subcategory} : {});
+
+    ctx.body = {products};
+  } catch (e) {
+    ctx.status = 400;
+    ctx.body = 'Bad Request';
+  }
+};
+
+module.exports.productList = async function productList(ctx, next) {
+  ctx.body = {};
+};
```

Comment:
I didn't understand what I had to do with this function

Reply:
After the task will be merged and accepted - please check the proposed solution. It shows how handler could be composed of small functions.